### DISCUSSION TITLE
I need help. Homematic Taster an Sonoff Actor.

### DISCUSSION THREAD

**Discussion by ghost** [thkl/hap-homematic#286] [Q&A]
When I add a switch (HM-PB-4-WM) to the Hap everything seems fine.

The switch is shown in the Home App and I can even make an automation with it. For instance: Press button an Kitchen light shall work. On the Iphone inside app no problem but there is no connection from the real switch to the button. Press the button on the Iphone. Everything ist fine. Press the real button and nothing. The switch even does not get white.

Where is my fault. Maybe someone could help me.

<img width="1127" alt="Bildschirmfoto 2020-12-21 um 23 42 14" src="https://user-images.githubusercontent.com/34451092/102829238-f365b480-43e6-11eb-8ac7-052b66e5dd06.png">

**Comment by thkl** [comment]
This Service is a "oneWay" It will only generate a PRESS_xxx Event and will not react on Key Events. I you want to trigger HomeKit Events by pressing a CCU Button you should use the HomeMaticKeyAccessory.

This is due to the fact that HomeKit does not provide a Service which can do both. There is a Programmable Switch (which will trigger HomeKit Events) or a On/Off Switch (which can take user inputs)
